Applications/utils/mem_usage.sh

```diff
@@ -10,14 +10,34 @@
 
 "$@" &
 process_id=$!
+peak=0
+mem_peak=0
+log="\nPID - $process_id :"
+outlog=""
+count=0
+while true; do
+    if [ -d "/proc/$process_id" ]; then
+	output=`grep VmHWM /proc/$process_id/status`
+	IFS=' ' read -ra mem_str <<< "$output"
+	count=$((count + 1))
+	mem_usage=$((${mem_str[1]}))
+	outlog="$outlog$count $mem_usage kB\n"
```

Comment:
later you can consider printing the log `$count $mem_usage kB\n` here itself to give user some feedback while the script progresses.